File: YemenSchoolsV1.Domain/Entities/SectionSubject.cs
namespace YemenSchoolsV1.Domain.Entities
{
    public class SectionSubject
    {
        public Guid Id { get; set; }
        public Guid SectionId { get; set; }
        public Guid GradeSubjectId { get; set; }
        public Guid TermId { get; set; }
        public Guid? TeacherId { get; set; }
        // Navigation property 
        public Section Section { get; set; } = null!;
        public GradeSubject GradeSubject { get; set; } = null!;
        public Term Term { get; set; } = null!;
        public Teacher Teacher { get; set; } = null!;
        public ICollection<Mark> Marks { get; set; } = [];
        public ICollection<DailyLog> DailyLogs { get; set; } = [];



    }
}

File: YemenSchoolsV1.Clint/src/app/features/school-dashboard/section-subject/models/section-subject.ts
export interface SectionSubject {
    id: string;
    sectionId: string;
    subjectId: string;
    termId: string;
    teacherId: string;
    teacherName?: string;
    subjectName?: string;
    termName?: string;
}

export interface CreateSectionSubjectDto {
    sectionId: string;
    gradeSubjectId: string;
    termId: string;
    teacherId: string;
}

export interface SectionSubjectUpdateDto {
    id: string;
    gradeSubjectId: string;
    termId: string;
    teacherId: string;
    sectionId: string;
}
